web/packages/teleterm/src/ui/Search/SearchContext.tsx

```diff
@@ -64,13 +64,6 @@ export const SearchContextProvider: FC = props => {
   const [isOpen, setIsOpen] = useState(false);
   const [inputValue, setInputValue] = useState('');
   const [activePicker, setActivePicker] = useState(actionPicker);
-  // TODO(ravicious): Consider using another data structure for search filters as we know that we
```

Comment:
I removed the comment because now we can select multiple resource filters, and I think the current structure works quite well.

Reply:
We could have an object with fields `clusterFilter` and `resourceFilters`, this way you wouldn't need to do the filtering based on the type field. From what I see, 90% of the type we end up either plucking the cluster filter from the array or filtering it down just resource filters.

There rest are places which have logic like this:

https://github.com/gravitational/teleport/blob/97c3757f3f9b19f9671f9d9362ad84bd8b1d00fc/web/packages/teleterm/src/ui/Search/useSearch.ts#L369-L371

It could be encapsulated in functions which work on the `SearchFilters` type (which would be that object with the fields I mentioned), here we could have something like `hasActiveSearchFilters(filters: SearchFilters): boolean`.

---

…aaand while writing this comment I realized what was the original requirement behind using an array – we need to keep the order of them so that if you first enter a resource filter and then a cluster filter, they're rendered in the UI in the same way.

Let's keep them this way I guess.

Reply:
Yeah, the requirement to keep the filters in order didn't allow me to change the structure.